lib/ecmascript.ts

```diff
@@ -1180,56 +1201,88 @@ type AllKeys<U> = Keys<U extends unknown ? U : never>;
 type AnyTemporalKey = Exclude<AllKeys<AnyTemporalLikeType>, symbol>;
 type Keys<T> = T extends Record<string, unknown> ? keyof T : never;
 type Values<T> = T[keyof T];
-type Entry<K> = K | readonly [K, (string | number | undefined)?];
-type ResolvedFieldTypes<T> = T extends Entry<infer K> ? K : never;
 type Owner<T extends AnyTemporalKey> = Values<{
   [N in keyof AllTemporalLikeTypes]: [T] extends [keyof AllTemporalLikeTypes[N]] ? AllTemporalLikeTypes[N] : never;
 }>;
-
 interface FieldPrepareOptions {
   emptySourceErrorMessage: string;
 }
 
+// UnionToIntersection. See https://fettblog.eu/typescript-union-to-intersection/
+type UnionToIntersection<U> = (
+  U extends any
+    ? // TS converts a union to an intersection because types in function
+      // arguments are contravariant. We extract this type using infer and return.
+      (k: U) => void
+    : never
+) extends (k: infer I) => void
+  ? I
+  : never;
```

Comment:
I managed to re-work this to avoid using UnionToIntersection, but I've kept this in mind and made more use of unknown over never.

I did use never in Owner/OwnerOf, and I'm wondering whether there's a difference there and those should also be unknown?

Reply:
Clarified where using `never` is critical / correct, vs where it was being used as an impossible error state by changing the latter to use string literals (which would cause all sorts of errors).